src/userInterface/components/JsonEditorContainer/JsonEditorContainer.tsx

```diff
@@ -12,38 +12,42 @@ import { DataType } from "../../../validator/enum/DataType.enum";
 
 const createValidInsertMenu = (submenu: MenuItem[] | undefined, currentJson: any, parentPath: (string | number)[]) => {
     const validMenuItems: MenuItem[] = [];
-    const validInsertItems: any = getValidInsertItems(parentPath);
-    const existingKeys: (number | string)[] = getExistingKeys(currentJson, [...parentPath]);
     const resultNode = addNode([...parentPath]).resultNode;
 
+    const validInsertItems: any = getValidInsertItems(parentPath, currentJson, resultNode);
+    const existingKeys: (number | string)[] = getExistingKeys(currentJson, [...parentPath]); 
+
     if (submenu === undefined || submenu.length === 0) {
         return undefined;
     }
 
     submenu?.forEach(subItem => {
         if (validInsertItems !== undefined && validInsertItems.length !== 0) {
-            let matchingItemCountWithSameDesc = 0;
+            // let matchingItemCountWithSameDesc = 0;
 
             Object.keys(validInsertItems).forEach((key: any) => {
-                if (subItem.text === key &&
-                    subItem.title === validInsertItems[key].description) {
+                if ((subItem.text === key)
+                    && (subItem.title === validInsertItems[key].description)
+                    && !existingKeys.includes(key)) {
                     /**
                      * filter alredy added items from insert menu
                      * unless it's map
                      */
-                    if (!(resultNode instanceof AdditionalNode) && !existingKeys.includes(key)) {
-                        validMenuItems.push(subItem);
-                    }
-                    if (resultNode instanceof AdditionalNode) {
+                    // if (!(resultNode instanceof AdditionalNode)) {
+                    //     validMenuItems.push(subItem);
+                    //     existingKeys.push(key);
+                    // }
+                    // if(resultNode instanceof AdditionalNode) {
                         validMenuItems.push(subItem);
-                    }
-                    matchingItemCountWithSameDesc++;
+                        existingKeys.push(key);
+                    // }
+                    // matchingItemCountWithSameDesc++;
                 }
             });
 
-            if (matchingItemCountWithSameDesc > 1) {
-                alert("non-unique Menu Items, Please use the valid option");
-            }
+            // if (matchingItemCountWithSameDesc > 1) {
```

Comment:
we cannot assume that schema only has unique keys and unique descriptions. isn't it?